rust/query/typeql_fetch.rs

```diff
@@ -147,24 +149,22 @@ impl<T: Into<Variable>> From<T> for ProjectionKeyVar {
 }
 
 #[derive(Debug, Eq, PartialEq)]
-pub enum ProjectionKeyLabel {
-    Quoted(String),
-    Unquoted(String),
+pub struct ProjectionKeyLabel {
+    pub label: String,
 }
```

Comment:
`ProjectionKeyLabel` is only quoted on `to_string()` and only when it has to be.

Reply:
This is actually not quite the intent - the user can choose to quote things even if they don't have to be and we're trying to remember which way they wrote the syntax. The `parser` always used quoted if the user wrote quotes. 

The 'choosing' of whether we have to quote or not only happens in the programmatic API where the user doesn't have to quote their strings (obviously!)

Reply:
We need to pick a lane here. Either we parse to have a faithful representation of the user input, or we parse to have a consistent internally usable representation of the user's query. We are currently stuck in the middle, and I would argue that we should stick to the latter, as the main two purposes for the typeql library are to parse the queries on the server side, and to build the queries on the client side, neither of which should care about faithfully representing the raw input.

Here, not having to worry about whether the user happened to quote the label saves us a match on the enum anywhere we might care to get that label. We can simplify a lot of things like these, and the benefits will add up.